salt/utils/templates.py

```diff
@@ -207,7 +207,8 @@ def render_tmpl(tmplsrc,
                 output = output.encode(SLS_ENCODING)
             if salt.utils.platform.is_windows():
                 newline = False
-                if output.endswith(('\n', os.linesep)):
+                # Use repr as we don't care about the contents, only the ending
+                if repr(output).endswith(('\n', os.linesep)):
```

Comment:
Are we sure this works?

```
>>> repr('foo\n')
"'foo\\n'"
>>> repr('foo\n').endswith(('\n'))
False
>>> 'foo\n'
'foo\n'
>>> 'foo\n'.endswith(('\n'))
True
```

Reply:
Yeah, it doesn't work... Let me see if I can figure out something else...